hook_event_dispatcher.module

```diff
@@ -221,6 +221,11 @@ function hook_event_dispatcher_preprocess(&$variables, $hook) {
   /* @var \Drupal\hook_event_dispatcher\Service\PreprocessEventService $service */
   $service = \Drupal::service('preprocess_event.service');
   $service->createAndDispatchKnownEvent($hook, $variables);
+
+  if ($service->isEntityHook($variables)) {
```

Comment:
Why not put all this stuff in the createAndDispatchKnownEvent and rename that to createAndDispatchKnownEvents? I try to keep the module file as clean as possible so the class does all the work 🙂.

Reply:
Oke will do so. I've got a question about the NodePreprocessEvent and the ParagraphPreprocessEvent. Since they are both content entities, they will now have their previous event and 2 extra event (preprocess_entity.node , preprocess_entity.paragraph). Should I remove these classes so all entities are handled the same, or would you rather keep them to not break backwards compatibility?

Reply:
Yeah was thinking about that as well, this is something we should fix with backwards compatibility in mind. Is there not a way to send out the more specific Node & Paragraph entity and just extend them from your new entity event/variables?

Reply:
I can do something like this :     
`$factory = $this->mapper->getFactory($hook) ? $this->mapper->getFactory($hook) : $this->mapper->getFactory('entity');` and only fire **preprocess_entity.node** when no specific factory is defined. 

So for a node of the type article it will fire **preprocess_node** and **preprocess_entity_bundle.node.article**

And for a unknown entity it will fire **preprocess_entity.MY_ENTITY** and **preprocess_entity_bundle.MY_ENTITY.BUNDLE**

And than make Event/Variables extend the new entity classes. So NodeEventVariables will also have a getEntity, ... method.

I think that would fix backwards compatibility since the event names will stay the same. It will only cause the event name to be inconsistent.